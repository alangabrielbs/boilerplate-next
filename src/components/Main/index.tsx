const Main = () => (
  <main>
    <h1>Titulo teste</h1>
  </main>
)

export default Main
