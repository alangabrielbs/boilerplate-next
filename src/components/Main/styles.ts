import styled from 'styled-components'

export const Wrapper = styled.main`
  background-color: #003ecb;
  color: #fff;
  height: 100%;

  padding: 3rem;

  display: flex;
  align-items: center;
  flex-direction: column;
  justify-content: center;
  text-align: center;
`

export const Title = styled.h1`
  font-size: 2.5rem;
  margin-bottom: 0.5rem;
`

export const Description = styled.h2`
  font-size: 2rem;
  font-weight: normal;
  margin-bottom: 3rem;
`

export const Illustration = styled.img`
  width: min(50rem, 90%);
`
