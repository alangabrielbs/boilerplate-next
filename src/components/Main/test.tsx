import { render, screen } from '@testing-library/react'

import Main from '.'

describe('<Main />', () => {
  it('Should render the heading', () => {
    render(<Main />)

    expect(
      screen.getByRole('heading', { name: /Titulo teste/i })
    ).toBeInTheDocument()
  })

  it('Should render correctly', () => {
    const { container } = render(<Main />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
