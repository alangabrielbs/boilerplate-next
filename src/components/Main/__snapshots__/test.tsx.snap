// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> Should render correctly 1`] = `
<main>
  <h1>
    Titulo teste
  </h1>
</main>
`
